server/src/logic/utils/error.utils.ts

```diff
@@ -7,4 +9,16 @@ export class ErrorUtils {
   static notFound (err: any): boolean {
     return err.code === errorCode.noData;
   }
+
+  static handleError(error: any, source: string, sourceMethod: string): ApiError{
```

Comment:
i like this error creation more than the long catch blocks. what did you think about use the builder pattern?